Sources/SwiftParser/Attributes.swift

```diff
@@ -869,6 +876,152 @@ extension Parser {
+      default: return nil
+      }
+    }
+
+    var rawTokenKind: RawTokenKind {
+      switch self {
+      case .branch: return .keyword(.branch)
+      case .from: return .keyword(.from)
+      case .revision: return .keyword(.revision)
+      }
+    }
+  }
+
+  mutating func parsePackageAttributeArguments() -> RawPackageAttributeArgumentsSyntax {
+    // Parsing package location.
+    let locationLabel = self.consume(ifAnyIn: PackageLocationLabel.self) ?? missingToken(.identifier)
+    let (unexpectedBeforeLocationColon, locationColon) = self.expect(.colon)
+    let (unexpectedBeforeLocation, location) = self.expect(.stringLiteral)
+    // Parsing package requirement.
+    let (unexpectedBeforeLocReqComma, locReqComma): (RawUnexpectedNodesSyntax?, RawTokenSyntax?)
+    let (unexpectedBeforeRequirementLabel, requirementLabel): (RawUnexpectedNodesSyntax?, RawTokenSyntax?)
+    let (unexpectedBeforeRequirementColon, requirementColon): (RawUnexpectedNodesSyntax?, RawTokenSyntax?)
+    let requirement: RawExprSyntax?
+    if locationLabel.tokenKind != .keyword(.path) {
+      (unexpectedBeforeLocReqComma, locReqComma) = self.expect(.comma)
+      if let label = self.consume(ifAnyIn: PackageRequirementLabel.self) {
+        (unexpectedBeforeRequirementLabel, requirementLabel) = (nil, label)
+        (unexpectedBeforeRequirementColon, requirementColon) = self.expect(.colon)
+      } else {
+        (unexpectedBeforeRequirementLabel, requirementLabel) = (nil, nil)
+        (unexpectedBeforeRequirementColon, requirementColon) = (nil, nil)
+      }
+      requirement = self.parseExpression()
+    } else {
+      (unexpectedBeforeLocReqComma, locReqComma) = (nil, nil)
+      (unexpectedBeforeRequirementLabel, requirementLabel) = (nil, nil)
+      (unexpectedBeforeRequirementColon, requirementColon) = (nil, nil)
+      requirement = nil
+    }
+    guard self.at(.comma) else {
+      return RawPackageAttributeArgumentsSyntax(
+        locationLabel: locationLabel,
+        unexpectedBeforeLocationColon,
+        locationColon: locationColon,
+        unexpectedBeforeLocation,
+        location: location,
+        unexpectedBeforeLocReqComma,
+        locReqComma: locReqComma,
+        unexpectedBeforeRequirementLabel,
+        requirementLabel: requirementLabel,
+        unexpectedBeforeRequirementColon,
+        requirementColon: requirementColon,
+        requirement: requirement,
+        reqProdComma: nil,
+        productLabel: nil,
+        productColon: nil,
+        productName: nil,
+        self.remainingTokensIfMaximumNestingLevelReached(),
+        arena: self.arena
+      )
+    }
+    let (unexpectedBeforeReqProdComma, reqProdComma) = self.expect(.comma)
+    let (unexpectedBeforeProductLabel, productLabel) = self.expect(.keyword(.product))
+    let (unexpectedBeforeProductColon, productColon) = self.expect(.colon)
+    let (unexpectedBeforeProductName, productName) = self.expect(.stringLiteral)
```

Comment:
Just a heads-up: This will need to be `parseStringLiteral` and you need to change the type of `product` to `StringLiteralExpr` in `AttributeNodes.py once https://github.com/apple/swift-syntax/pull/1192 is merged.